### PR TITLE
feat: add tauri-schema-worker

### PR DESCRIPTION
**Pull request by amrbashir**
ref: https://github.com/tauri-apps/create-tauri-app/issues/725#issuecomment-2185859045

### CODE REVIEW

File: crates/tauri-schema-worker/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+use anyhow::Context;
```
Comment (amrbashir) [inline crates/tauri-schema-worker/src/config.rs:5]:
```suggestion
// Copyright 2019-2024 Tauri Programme within The Commons Conservancy
// SPDX-License-Identifier: Apache-2.0
// SPDX-License-Identifier: MIT

use anyhow::Context;
```

File: .github/workflows/deploy-schema-worker.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2019-2022 Tauri Programme within The Commons Conservancy
```
Comment (amrbashir) [inline .github/workflows/deploy-schema-worker.yml]:
```suggestion
# Copyright 2019-2024 Tauri Programme within The Commons Conservancy
```

File: .changes/config.json [hunk 3]
```diff
@@ -212,16 +203,16 @@
         "tauri-runtime-wry",
         "tauri-build"
       ],
-      "postversion": "node ../../.scripts/ci/sync-cli-metadata.js ${ pkg.pkg } ${ release.type }"
+      "postversion": [
+        "node ../../.scripts/ci/sync-cli-metadata.js ${ pkg.pkg } ${ release.type }",
+        "cargo build --manifest-path ../tauri-schema-generator/Cargo.toml"
```
Comment (lucasfernog) [inline .changes/config.json:208]: should probably do this only when tauri-utils bumps instead

Comment (lucasfernog) [reply .changes/config.json:208]: nvm when utils bump tauri bumps too anyway.. and the schema is tied to the tauri version

File: crates/tauri-schema-worker/src/config.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  let Some(version) = releases.into_iter().find(|r| version.matches(&r.version)) else {
+    return max_stable_schema().await;
+  };
+
+  schema_file_for_version(version.version).await
+}
+
+async fn crate_releases(crate_: &str) -> anyhow::Result<Vec<CrateRelease>> {
```
Comment (lucasfernog) [inline crates/tauri-schema-worker/src/config.rs]: what do you think about caching these for a while? maybe in the future even add a webhook for tauri publishes?

File: .github/workflows/deploy-schema-worker.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: cloudflare/wrangler-action@3
+        with:
+          command: deploy
+          workingDirectory: 'packages/schema-worker'
```
Comment (lucasfernog) [inline .github/workflows/deploy-schema-worker.yml]: isn't this in the crates folder?

Comment (amrbashir) [reply .github/workflows/deploy-schema-worker.yml]: yeah that's correct, I initially wrote the worker in JS but later switched to rust so the directory changed

File: crates/tauri-schema-worker/src/config.rs (hunk 4 above)
Comment (amrbashir) [reply crates/tauri-schema-worker/src/config.rs]: that's a good idea, will have to read more about workers 😅

Comment (lucasfernog) [reply crates/tauri-schema-worker/src/config.rs]: maybe the worker can cache the JSON schema per version, then when we request `/config/2` we just redirect to the latest version `/config/2.1.2` or whatever, so we don't really need to deal with any cache on our end
David knows how to do it, he wrote some cloudflare caching in the past

Comment (amrbashir) [reply crates/tauri-schema-worker/src/config.rs]: `/config/2` is resolved to `2.1.2` or whatever latest is, and response will be cached for resolved version so `/config/2` or `/config/2.1` or `/config/2.1.2` or `/config/latest` (and other aliases) will use the same file from cache.

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
this is just a beauty

**Pull request merged**